### PR TITLE
Disallows nested headings when there is no content between them

### LINKED ISSUE #257: Nested heading immediately after a heading of the same level incorrectly passes
## Summary
Nested heading immediately after a heading of the same level incorrectly passes the "Structure: Headings: Content must follow headings" guideline

## Expected Behaviour
The following markup should cause the "Structure: Headings: Content must follow headings" guideline to fail, as there is no content between the two `h2` elements:
```      
      <h1>Main heading</h1>
      <h2>This is a sub-heading</h2>
      <div>
        <h2>Another h2 heading</h2>
        <p>Hello</p>
      </div>
```
## Current Behaviour
The guideline passes

## Possible Solution
Fix the logic for this guideline

### PR DESCRIPTION
**Pull request by andymsuk**
## Summary
Fixes #257

## Details
See #257

## How Has This Been Tested?
Added more Feature scenarios

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/standards/tests/contentMustFollowHeadings.js [hunk 1]
```diff
@@ -16,18 +19,44 @@ module.exports = {
...
+          })
+        }
+        if (!nodeStatus) {
+          nodeStatus = checkNextNode(heading, $nextNode)
+        }
+        if (nodeStatus === NODE_IS_FINE) {
+          return
+        }
+        if (nodeStatus === NODE_IS_BAD) {
+          break
```
[the remark this answers is not in the document]
Comment (andymsuk) [reply lib/standards/tests/contentMustFollowHeadings.js]: Not quite. A `return` will return out of that iteration of the `$(headingSelector).each` and continue on to the next heading.

The `break` will drop down into the `fail`.

We did sit and stare at it for a while and couldn't find an easy way to make it neater. Want to have a go?

[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply lib/standards/tests/contentMustFollowHeadings.js]: Having a go here: https://github.com/bbc/bbc-a11y/pull/278

Harder than I thought! So many edge cases 😬in light of this, happy to approve your PR!

**Comment by ChrisBAshton** [comment]
Released v2.2.6 and updated the internal bbc-a11y-maintenance doc.
